src/Operation/Aggregate.php

```diff
@@ -0,0 +1,159 @@
+     *    necessary to aid in mongod/mongos upgrades.
+     *
+     * @param string $collectionName
+     * @param array  $pipeline       List of pipeline operations
+     * @param array  $options        Command options
+     * @throws InvalidArgumentException
+     */
+    public function __construct($collectionName, array $pipeline, array $options = array())
+    {
+        $options += array(
+            'allowDiskUse' => false,
+            'useCursor' => true,
+        );
+
+        if ($options['allowDiskUse'] && ! is_bool($options['allowDiskUse'])) {
+            throw new InvalidArgumentTypeException('allowDiskUse option', $options['allowDiskUse'], 'boolean');
+        }
+
+        if (isset($options['batchSize']) && ! is_integer($options['batchSize'])) {
+            throw new InvalidArgumentTypeException('batchSize option', $options['batchSize'], 'integer');
+        }
+
+        if (isset($options['maxTimeMS']) && ! is_integer($options['maxTimeMS'])) {
+            throw new InvalidArgumentTypeException('maxTimeMS option', $options['maxTimeMS'], 'integer');
+        }
+
+        if ( ! is_bool($options['useCursor'])) {
+            throw new InvalidArgumentTypeException('useCursor option', $options['useCursor'], 'boolean');
+        }
+
+        if (isset($options['batchSize']) && ! $options['useCursor']) {
+            throw new InvalidArgumentException('batchSize option should not be used if useCursor is false');
+        }
+
+        $expectedIndex = 0;
+
+        foreach ($pipeline as $i => $op) {
+            if ($i !== $expectedIndex) {
+                throw new InvalidArgumentException(sprintf('$pipeline is not a list (unexpected index: "%s")', $i));
+            }
+
+            if ( ! is_array($op) && ! is_object($op)) {
+                throw new InvalidArgumentTypeException(sprintf('$pipeline[%d]', $i), $op, 'array or object');
+            }
+
+            $expectedIndex += 1;
+        }
+
+        $this->collectionName = $collectionName;
+        $this->pipeline = $pipeline;
+        $this->options = $options;
+    }
+
+    /**
+     * Execute the operation.
+     *
+     * @see MongoDB\Collection::createIndexes()
+     * @see http://php.net/bson-serializable.bsonserialize
+     */
+    public function execute(Server $server, $databaseName)
+    {
+        $command = $this->createCommand($server);
+        $cursor = $server->executeCommand($databaseName, $command);
+
+        if ($this->options['useCursor']) {
+            return $cursor;
+        }
+
+        $result = current($cursor->toArray());
+
+        if (empty($result['ok'])) {
+            throw new RuntimeException(isset($result['errmsg']) ? $result['errmsg'] : 'Unknown error');
+        }
+
+        return new ArrayIterator(array_map(
+            function (\stdClass $document) { return (array) $document; },
+            $result['result']
+        ));
+    }
+
+    /**
+     * @param boolean $isCursorSupported
```

Comment:
This was leftover from when the wire version check was in `execute()` (before I moved it to `createCommand()`).